Plot within batch ARI

allyhawkins opened the pull request:
Closes #203 

Here I added a plot for calculating the within batch ARI to the integration report. To be able to add this in I had to also add a way to first calculate the within batch ARI which requires having the individual SCE objects for all the libraries present in the integrated objects being evaluated. To do that I decided to add the library metadata file as a parameter. Then from there we can use the `integration_input_dir` column to define the directory that holds the original SCE objects and find them by filtering to the library IDs that are associated with the `params$group` being tested. I added the chunk that does the reading in of the individual files and creation of the list of individual SCE files towards the beginning of the notebook, after reading in the integrated data since it seemed to fit with that. Also I added a check to make sure all the expected files are present so I wanted that near the top. 

Then after looking at batch and celltype ARI, I added this new ARI where I just made one simple plot with integration method on the x-axis and ari on the y-axis. I didn't really think we needed anything too fancy here. I'm including two examples of testing this with the simulated data. 

Another thing I did here was remove the `unintegrated` option and actually allowed for `unintegrated` to be an `integration_method`. It makes being able to apply our function across the sce list that we have in the notebook a lot simpler. If people have strong methods about that I can change it back. I also didn't write a separate function here to calculate and then plot, because it wasn't super complicated. And having to return to this repo after a little break, I like having the code within the notebook rather than trying to parse through layers of functions. 

Simulated data with shared cell types
![image](https://user-images.githubusercontent.com/54039191/228973174-662d8a6d-db0a-452f-9b3c-41c435fd7285.png)


Simulated data where not all cell types are shared across all batches 
![image](https://user-images.githubusercontent.com/54039191/228972616-29a7f3f7-bf36-498b-b004-40408790a0a6.png)

I'm going to tag both @jashapiro and @sjspielman and whoever gets to this first can take the review.

sjspielman commented on analysis_templates/01-single-group-integration-check-template.Rmd:
Leaving some initial comments here - At first I wondered why we can't just split up the unintegrated SCE object back into it's individual SCEs, rather than having to go back to each individual unintegrated file. My conclusion is this is because our merging _overwrites_ the PCs calculated from individual libraries, and those are the PCAs we want to use in these calculations. Just confirming here!

sjspielman replied:
Also, wanted to say - I think it would be helpful to add _small, short, very brief_ markdown sentence above each of these prep chunks that get hidden in the HTML. Something as quick as "Next, read in the unintegrated SCEs" above a chunk will make it easier to "unhide" given chunks of interest in the HTML.

allyhawkins replied:
> Leaving some initial comments here - At first I wondered why we can't just split up the unintegrated SCE object back into it's individual SCEs, rather than having to go back to each individual unintegrated file. My conclusion is this is because our merging overwrites the PCs calculated from individual libraries, and those are the PCAs we want to use in these calculations. Just confirming here!

That is correct. We overwrite them when we calculate the batch corrected PC. Additionally, we subset the genes based on shared genes that are present in the merged object, so we couldn't go back and back-calculate the original PCs either.

sjspielman commented on analysis_templates/01-single-group-integration-check-template.Rmd:
```diff
@@ -508,6 +550,56 @@ if (celltypes_present) {
+The goal of this metric is to measure how well a given integration method preserves biological heterogeneity by comparing clustering assignments from pre integration to post integration for each library. 
+This function takes each original SCE object, prior to any merging, and performs graph based clustering. 
+The same clustering is then performed on the integrated object.
+For each library that is present in the integrated object, the original cluster assignments are compared to the new cluster assignments for those cells post integration by calculating the ARI between the cluster assignments. 
```
```suggestion
The original cells' cluster assignments calculated on each individual library in the integrated object are then compared to the cells' new cluster assignments post-integration by comparing ARI between these two cluster assignments. 
```

sjspielman reviewed (commented):
Seems mostly good to me! I left some text suggestions, as well as one conceptual comment to confirm the approach here.

allyhawkins commented:
I incorporated most of the suggestions that were made. As far as adding in small markdown comments in front of hidden code chunks, I added a few in front of the larger chunks that are more important but not all of them. I don't know if adding them to all the chunks is necessary for internal reports like this. This should be ready for another look.

jashapiro commented on analysis_templates/01-single-group-integration-check-template.Rmd:
```diff
@@ -508,6 +554,56 @@ if (celltypes_present) {
+within_batch_ari <- set_integration_order(within_batch_ari)
+
+# plot within batch ARI by integration method
+ggplot(within_batch_ari, aes_string(x = "integration_method_factor" , y = "ari")) +
```
> * for example:
> 
> ```
> a_real_function <- function(numeric_iris_column) {
>   ggplot(iris, aes(x = {{numeric_iris_column}})  + geom_histogram()
> }
> 
> # Usage is **without quotes**
> a_real_function(Sepal.Length)
> ```

I assume you meant something more like:

```
a_real_function <- function(data, numeric_column) {
  ggplot(data, aes(x = {{numeric_column}})  + geom_histogram()
}

a_real_function(iris, Sepal.Length)
```

Because I _know_ you would't use external variables in a function.

sjspielman replied:
I want everyone to know, and this is true, that I wrote that example intentionally without `iris` to see if it would elicit this exact comment, so this is an all-around team success 🎉 🎉 

Of course, I would _never_ do such a thing in production 😄

sjspielman reviewed (approved):
🚀
